sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/HttpHandler.cs

```diff
@@ -136,15 +138,35 @@ private static void CompleteFailedRequest(IHttpRequest<TRequestContent> httpRequ
             catch { }
         }
 
-        private static void SetTraceSpanHttpAttributes(TraceSpan traceSpan, string method, IRequest request, IWebResponseData response)
+        private static void RecordHttpTelemetryData(IExecutionContext executionContext, TraceSpan traceSpan, IRequest request)
         {
-            traceSpan.SetAttribute(TelemetryConstants.HTTPMethodAttributeKey, method);
+            var response = executionContext.ResponseContext.HttpResponse;
+
+            var metricsAttributes = new Attributes();
+            metricsAttributes.Set(TelemetryConstants.ServerAddressAttributeKey, $"{request.Endpoint.Host}:{request.Endpoint.Port.ToString()}");
+
+            traceSpan.SetAttribute(TelemetryConstants.HTTPMethodAttributeKey, request.HttpMethod);
             traceSpan.SetAttribute(TelemetryConstants.HTTPStatusCodeAttributeKey, ((int)response.StatusCode));
 
             var contentLengthHeader = request.GetHeaderValue(HeaderKeys.ContentLengthHeader);
-            if (!string.IsNullOrEmpty(contentLengthHeader) && long.TryParse(contentLengthHeader, out var contentLength))
+            if (long.TryParse(contentLengthHeader, out var contentLength) && contentLength > 0)
```

Comment:
What is the reason of treating a missing content-length the same as one that is zero? Is this correct in all cases or what case supports this change?